docs/CONTRIBUTING.md

```diff
@@ -1,113 +1,972 @@
+
+    We currently use **AGP 8.1.0** which requires developers to use **JDK 17** on Gradle JDK.
+
+    _If you're on macOS, you can install it with [brew](https://formulae.brew.sh/formula/openjdk@17)_
+
+    ```bash
+    brew install openjdk@17
+    ```
+
+    > ℹ️ If you're using a device with [Apple silicon (M1/M2)](https://support.apple.com/en-us/HT211814) then you might need to install
+    > a [zulu distribution](https://www.azul.com/downloads/zulu-community/?version=java-17-lts&architecture=x86-64-bit&package=jdk)
+
+    Alternatively, install it directly from **Android Studio**:  
+    `File` → `Settings` → `Build, Execution, Deployment` → `Build Tools` → `Gradle` → `Gradle JDK`
+
+---
+
+## Project Setup
+
+Once you have a compatible environment as described above, you can set up the project:
+
+1. **Clone the repository**
+
+   ```bash
+   git clone https://github.com/adevinta/spark-android.git
+   cd spark-android
+   ```
+
+2. **Open the project in Android Studio**
+   - File > Open > Select the `spark-android` directory
+
+3. **Build the project**
+
+   ```bash
+   ./gradlew build
+   ```
+
+---
+
+## 🧩 Creating Components
+
+The Spark Design System follows a standardized architecture for all components. This section walks you through creating a new component from scratch.
+
+> [!NOTE]
+> This section contains all the architectural standards and patterns you need to follow.
+
+### Component Structure
+
+Each component follows this standardized file structure:
+
+```text
+spark/src/main/kotlin/com/adevinta/spark/components/{component-name}/
+├── Component.kt                 # Public API surface
+├── ComponentDefaults.kt         # Design token mappings and default values
+├── ComponentIntent.kt           # Semantic color variants
+├── ComponentSize.kt            # Size specifications
+├── ComponentState.kt           # Behavioral state management (if needed)
+└── Component.md                # API documentation and usage guidelines
+```
+
+### API Design Principles
+
+All component APIs must follow this parameter ordering convention:
+
+```kotlin
+@Composable
+public fun ComponentName(
+    // 1. Required behavioral parameters
+    onClick: () -> Unit,
+    
+    // 2. Standard Compose parameters
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    
+    // 3. Design system parameters (semantic abstractions)
+    intent: ComponentIntent = ComponentIntent.Basic,
+    size: ComponentSize = ComponentSize.Medium,
+    
+    // 4. Advanced behavioral parameters
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+
+    // 5. Content slots (always last)
+    content: @Composable () -> Unit,
+)
+```
+
+### Example: Creating a Badge Component
+
+<details>
+<summary>Complete Badge Component Implementation</summary>
+
+1. **Create the main component file** (`spark/src/main/kotlin/com/adevinta/spark/components/badge/Badge.kt`):
```

Comment:
I am pretty sure nobody is going to read this example. Could we instead simply link to the `spark/src/main/kotlin/com/adevinta/spark/components/badge` directory for a simple and complete example?